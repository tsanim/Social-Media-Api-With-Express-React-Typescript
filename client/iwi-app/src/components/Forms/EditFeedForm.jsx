import React from 'react';
import useForms from '../../hooks/useForms';
import PropTypes from 'prop-types';

function EditFeedForm({ feedId, text, editUserPostHandler, handleShowEditForm }) {
    const { handleSubmit, handleChangeInput, inputs } = useForms((e) => {
        editUserPostHandler(inputs, feedId);
        handleShowEditForm();
    });

    return (
        <form id="editFeedForm" onSubmit={handleSubmit}>
            <textarea
                placeholder="Write something..."
                name="text"
                id="feed"
                value={inputs.text || text}
                onChange={handleChangeInput}
            />
            <input type="submit" value="SAVE" />
            <button onClick={handleShowEditForm}>Undo</button>
        </form>
    )
}

EditFeedForm.propTypes = {
    feedId: PropTypes.string,
    text: PropTypes.string,
    editUserPostHandler: PropTypes.func,
    handleShowEditForm: PropTypes.func,
}

export default EditFeedForm;