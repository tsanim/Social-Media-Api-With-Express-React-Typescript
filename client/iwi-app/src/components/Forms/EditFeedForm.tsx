import React from 'react';
import useForms from '../../hooks/useForms';
import PropTypes from 'prop-types';
import EditFeedFormProps from '../../interfaces/Components/Forms/EditFeedFormProps.interface';

function EditFeedForm({ feedId, text, editUserPostHandler, handleShowEditForm }: EditFeedFormProps) {
    const { handleSubmit, handleChangeInput, inputs } = useForms(() => {
        editUserPostHandler(inputs, feedId);
        handleShowEditForm();
    });

    return (
        <form id="editFeedForm" onSubmit={handleSubmit}>
            <textarea
                placeholder="Write something..."
                name="text"
                id="feed"
                value={inputs.text || text}
                onChange={handleChangeInput}
            />
            <input type="submit" value="SAVE" />
            <button onClick={handleShowEditForm}>Undo</button>
        </form>
    )
}

EditFeedForm.propTypes = {
    feedId: PropTypes.string,
    text: PropTypes.string,
    editUserPostHandler: PropTypes.func,
    handleShowEditForm: PropTypes.func,
}

export default EditFeedForm;