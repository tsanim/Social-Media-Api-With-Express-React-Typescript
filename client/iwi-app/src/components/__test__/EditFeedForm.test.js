import React from 'react';
import { shallow } from 'enzyme';

import EditFeedForm from '../Forms/EditFeedForm';

function editFeedFormComponentSetup() {
    const props = {
        feedId: '',
        text: undefined,
        editUserPostHandler: jest.fn(),
        handleShowEditForm: jest.fn(),
    }

    const enzymeWrapper = shallow(<EditFeedForm {...props} />)

    return {
        props,
        enzymeWrapper
    }
}


describe('EditFeedForm Component', () => {
    it('should render initial layout', () => {
        const { enzymeWrapper } = editFeedFormComponentSetup();

        expect(enzymeWrapper.getElements()).toMatchSnapshot();
    });

    it('should render form tag with id "editFeedForm"', () => {
        const { enzymeWrapper } = editFeedFormComponentSetup();

        expect(enzymeWrapper.find('form#editFeedForm')).toHaveLength(1);
    });

    it('should render textarea tag', () => {
        const { enzymeWrapper } = editFeedFormComponentSetup();

        expect(enzymeWrapper.find('textarea')).toHaveLength(1);
    });

    it('should render textarea tag with name prop "text"', () => {
        const { enzymeWrapper } = editFeedFormComponentSetup();

        expect(enzymeWrapper.find('textarea').prop('name')).toEqual('text');
    });

    it('should render textarea tag with onChange prop', () => {
        const { enzymeWrapper } = editFeedFormComponentSetup();

        expect(enzymeWrapper.find('textarea').prop('onChange')).toBeDefined();
    });

    it('should render input tag with type "submit"', () => {
        const { enzymeWrapper } = editFeedFormComponentSetup();

        expect(enzymeWrapper.find('input[type="submit"]')).toHaveLength(1);
    });

    it('should render a button tag', () => {
        const { enzymeWrapper } = editFeedFormComponentSetup();

        expect(enzymeWrapper.find('button')).toHaveLength(1);
    });

    it('should test if user text is echoed', () => {
        const { enzymeWrapper } = editFeedFormComponentSetup();

        const mockedEvent = {
            target: { value: 'hello', name: 'text' },
            persist: jest.fn()
        };

        enzymeWrapper.find('textarea').simulate('change', mockedEvent);

        expect(enzymeWrapper.find('textarea').prop('value')).toEqual('hello');
    });

    it('should test if submit event when form is submitted', () => {
        const { enzymeWrapper, props } = editFeedFormComponentSetup();

        const mockedEvent = {
            preventDefault: () => {},
        };

        enzymeWrapper.find('form').simulate('submit', mockedEvent);

        expect(props.editUserPostHandler).toHaveBeenCalledTimes(1);
        expect(props.handleShowEditForm).toHaveBeenCalledTimes(1);
    });
});
