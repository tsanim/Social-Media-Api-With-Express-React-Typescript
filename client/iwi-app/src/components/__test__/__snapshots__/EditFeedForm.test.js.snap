// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditFeedForm Component should render initial layout 1`] = `
Array [
  <form
    id="editFeedForm"
    onSubmit={[Function]}
  >
    <textarea
      id="feed"
      name="text"
      onChange={[Function]}
      placeholder="Write something..."
    />
    <input
      type="submit"
      value="SAVE"
    />
    <button
      onClick={[MockFunction]}
    >
      Undo
    </button>
  </form>,
]
`;
